Sources/Map/Map+Coordinator.swift

```diff
@@ -74,12 +76,42 @@ extension Map {
         private func updateAnnotations(on mapView: MKMapView, from previousView: Map?, to newView: Map) {
             let changes: CollectionDifference<AnnotationItems.Element>
             if let previousView = previousView {
-                changes = newView.annotationItems.difference(from: previousView.annotationItems) { $0.id == $1.id }
+                changes = newView.annotationItems.difference(from: previousView.annotationItems) { $0 == $1 }
             } else {
-                changes = newView.annotationItems.difference(from: []) { $0.id == $1.id }
+                changes = newView.annotationItems.difference(from: []) { $0 == $1 }
             }
-
-            for change in changes {
+            
+            let changeElement = { (change: CollectionDifference<AnnotationItems.Element>.Change) -> AnnotationItems.Element in
+                switch change {
+                case let .insert(_, element, _):
+                    return element
+                case let .remove(_, element, _):
+                    return element
+                }
+            }
+            
+            let changeId = { change in
+                return changeElement(change).id
+            }
+            
+            let insertionSet = Set(changes.insertions.map(changeId))
+            let removalsSet = Set(changes.removals.map(changeId))
+            let modificationsSet = insertionSet.intersection(removalsSet)
+            // we're only looking at insertions to avoid duplicates
+            let modificationsList = changes.insertions.lazy.filter { modificationsSet.contains(changeId($0)) }.map(changeElement)
+            
+            for modification in modificationsList {
+                if
+                    let content = annotationContentByID[modification.id],
+                    let annotationView = viewByObject[ObjectIdentifier(content.annotation)],
+                    let annotation = content.annotation as? MKPointAnnotation
+                {
+                    view?.modifiedAnnotationHandler?(modification, annotation, annotationView)
+                }
+            }
```

Comment:
find a way to avoid doing all this if `modifiedAnnotationHandler` doesn't exist

Reply:
I might be able to help here - similarly to https://github.com/pauljohanneskraft/Map/issues/15 you might be able to create a publisher to change these properties.

You can use views like the following to create these publishers:

```swift
struct ChangeListener<Value: Equatable, Content: View>: View {

    // MARK: Stored Properties

    let value: Value
    @ViewBuilder var content: (PassthroughSubject<Value, Never>) -> Content

    @State private var subject = PassthroughSubject<Value, Never>()

    // MARK: Computed Properties

    var body: some View {
        content(subject)
            .onChange(of: value, perform: subject.send)
    }

    // MARK: Initialization

    init(for value: Value, @ViewBuilder content: @escaping (PassthroughSubject<Value, Never>) -> Content) {
        self.value = value
        self.content = content
    }

}
```

In many cases with these highly fluctuating and often-changing values, you will probably already have publishers (or async sequences) anyways in these cases, right? 😅

Reply:
Alternatively, you can also look into `ObservableObject` and how you might be able to use its `objectWillChangePublisher` that is triggered for each `@Published` property.

Reply:
Thanks for your feedback! I'm a little confused though. How do you see `ChangeListener` being used?

Reply:
I _think_ I get what you're going for actually. Create `ChangeListener`, use it to wrap the view being passed to a `ViewMapAnnotation`, and pass it a publisher that will be updated with a changing value so the wrapped view can be updated based on those changes.

Is that it? I've got to move on for now at work, but I'll come back to this PR at some point. Feel free to steal what's here if you want to, or close the PR in the meantime and I'll reopen it once I have time.